doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,422 @@
+         * GIDSet will become effective once the next read operation is
+         * completed.
+         *
+         * This operation may throw away cached data and make getCurrentTime()
+         * jump a step bigger than the timestep for the next read. It is
+         * implementation defined when this may happen.
+         *
+         * @throw std::runtime_error is the report was open in write mode or
+         *        there's an on-going read operation.
+         */
+        void updateMapping( const GIDSet& gids );
+
+        /** @return In read mode, return the GIDSet that applies to the result\
+         *          of the last read operation (this set comes from either the
+         *          constructor or a call to updateMapping). In write mode it
+         *          returns the GID set that has been set using writeMapping.
+         */
+        const GIDSet& getGIDs() const;
+
+        /** Get the current mapping of each section of each neuron in each
+         *  simulation frame buffer.
+         *
+         * For instance, getOffsets()[1][15] retrieves the lookup index for the
+         * frame buffer for section 15 of neuron with index 1. The neuron index
+         * is derived from the order in the GID set provided by updateMapping().
+         *
+         * For requested GIDs that are not part of the report its section offset
+         * list will be empty.
+         *
+         * After construction or updateMapping is called, the result is empty
+         * until a frame is read.
+         *
+         * @return the offset for each section for each neuron
+         */
+        const SectionOffsets& getOffsets() const;
+
+        /** Get the number of compartments for each section of each neuron
+         *  provided by the GID set via updateMapping().
+         *
+         * The neuron index is derived from the order in the GID set provided by
+         * updateMapping(). For requested GIDs that are not part of the report
+         * its compartment count list will be empty.
+         *
+         * After construction or updateMapping is called, the result is empty
+         * until a frame is read.
+         *
+         * @return the compartment counts for each section for each neuron
+         * @version 1.0
+         */
+        const CompartmentCounts& getCompartmentCounts() const;
+
+        /** Get the number of compartments for the given neuron.
+         *
+         * The neuron index is derived from the order in the GID set provided by
+         * updateMapping().
+         *
+         * @param index neuron index per current GID set
+         * @return number of compartments for the given neuron or 0 if the
+         *         report has just been created or updateMapping is called, and
+         *         not frame has been read yet. Also 0 for neurons not part of
+         *         the report at the source.
+         */
+        size_t getNumCompartments( const size_t index ) const;
```

Comment:
Is this the same as getCompartmentCounts()[index]? If so is it bringing much? It's also the only function which takes an index instead of a gid, slightly surprising for users.